iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2418,11 +2418,7 @@ static void ProcessPendingTelemetryMessages(PMQTTTRANSPORT_HANDLE_DATA transport
                 free(msg_detail_entry);
 
                 DisconnectFromClient(transport_data);
-                if (!transport_data->isRetryExpiredCallbackCalled) // Only call once
-                {
-                    transport_data->transport_callbacks.connection_status_cb(IOTHUB_CLIENT_CONNECTION_UNAUTHENTICATED, IOTHUB_CLIENT_CONNECTION_RETRY_EXPIRED, transport_data->transport_ctx);
-                    transport_data->isRetryExpiredCallbackCalled = true;
-                }
+                transport_data->transport_callbacks.connection_status_cb(IOTHUB_CLIENT_CONNECTION_UNAUTHENTICATED, IOTHUB_CLIENT_CONNECTION_COMMUNICATION_ERROR, transport_data->transport_ctx);
```

Comment:
just double checking here.....we no longer have the `if()` condition on calling it once since this is for a different reason?

Reply:
correct. the problem was that IOTHUB_CLIENT_CONNECTION_RETRY_EXPIRED was also fired in another place, so two events were fired with the same value.